### PR TITLE
Replacing context.ReportDiagnostic with context.CheckedReportDiagnost…

### PR DESCRIPTION
**Pull request by valhristov**
…ic extension method

In two words: created extension methods in AnalysisContextExtensions and replaced all calls to context.ReportDiagnostic with them. If you have better suggestions about the names of the methods I will gladly change :)

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ */
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace SonarAnalyzer.Helpers
+{
+    public static class AnalysisContextExtensions
+    {
+        public static void CheckReportDiagnostic(this SyntaxNodeAnalysisContext context, Diagnostic diagnostic)
```
Comment (duncanp-sonar) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs]: I guess _ReportDiagnosticIfNotSuppressedInSonarQube(...)_ is a bit long :-)

Alternative suggestions: _CheckAndReportDiagnostic_, _ReportCheckedDiagnostic_ _ReportIfNotSuppressed_?
I don't think any of the alternatives are much of an improvement.

**Comment by valhristov** [comment]
@michalb-sonar it will be challenging to add a simple test because I will have to make every rule execute and report at least once, which will mean to execute all tests one more time. It is doable, but I don't know if we want to double our execution time.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs (hunk 1 above)
Comment (valhristov) [reply sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs]: I was thinking of `CheckAndReportDiagnostic`, but I did not like the `And` in the name.

Comment (Evangelink) [reply sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs]: I quite like _ReportIfNotSuppressed_.

Comment (valhristov) [reply sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs]: I like `ReportCheckedDiagnostic`, also `ReportActiveDiagnostic` `ReportDiagnosticConditional` `ReportDiagnosticIfActive` `ReportDiagnosticWhenActive`... Ping @duncanp-sonar @Evangelink @michalb-sonar

**Comment by valhristov** [comment]
Changed the names of the extension method and the static DI method.

**Pull request merged**